### PR TITLE
Login with amazon added for both iOS and Android

### PR DESCRIPTION
**Pull request by ianbattin**
Pretty big addition, though most changes take place in this repo: https://github.com/ianbattin/react-native-login-with-amazon as this is where the native libraries live.

-Added the Login with Amazon button on the login page
-Added login actions to sign in using amazon
-Added Firebase Cloud Functions to support logging in with amazon

### CODE REVIEW

**Review by smrizzo — APPROVED** [review]
All looks good to me.

File: README.md [hunk 1]
```diff
@@ -46,3 +46,28 @@ or
...
 react-native run-android
 ```
+
+## Firebase Cloud Functions
+
+To contribute to our firebase cloud functions, you need to install some dependenices on your machine.
+
+The documentation is here: https://firebase.google.com/docs/functions/get-started
+
+Badically, to get set up, do the following:
```
Comment (BryceV) [inline README.md:56]: Typo on line 52 ("dependencies"), and 56("basically")

File: src/screens/Auth/Login.js [hunk 2]
```diff
@@ -10,18 +17,29 @@ export default class Login extends Component {
...
     this.createAccount = this.createAccount.bind(this)
     this.forgotPassword = this.forgotPassword.bind(this)
+    this.loginWithAmazon = this.loginWithAmazon.bind(this)
+  }
+
+  loginWithAmazon() {
+    this.setState({amazonWaiting: true})
+    this.props.amazonLogin().then(() => {
+      this.setState({amazonWaiting: false})
+      if(this.props.User)
```
Comment (ianbattin) [inline src/screens/Auth/Login.js:34]: maybe catch this instead

File: src/screens/Auth/actions.js [hunk 3]
```diff
@@ -1,4 +1,53 @@
 import firebase from 'react-native-firebase'
+import LoginWithAmazon from 'react-native-login-with-amazon'
+
+export function amazonLogin(email, password) {
+  return (dispatch, prevState) => {
+    return new Promise(async (resolve, reject) => {
+      LoginWithAmazon.login((error, accessToken, profileData) => {
+        if(error) {
+          resolve()
```
Comment (ianbattin) [inline src/screens/Auth/actions.js:9]: maybe reject instead

File: src/screens/Auth/Login.js (hunk 2 above)
Comment (abineetsingh) [reply src/screens/Auth/Login.js:34]: Ian made a custom React Native API for performing the login functionality which can be ported via npm which is quite impressive!

File: ios/AmazonHandmadeRN/Info.plist [hunk 4]
```diff
@@ -53,6 +52,10 @@
 			</dict>
 		</dict>
 	</dict>
+	<key>NSLocationWhenInUseUsageDescription</key>
+	<string/>
```
Comment (BryceV) [inline ios/AmazonHandmadeRN/Info.plist:56]: Is this tag used?

File: src/actions/index.js [hunk 5]
```diff
@@ -1,8 +1,15 @@
 import firebase from 'react-native-firebase'
+import LoginWithAmazon from 'react-native-login-with-amazon'
 
 export function authLogout() {
   return (dispatch, prevState) => {
     return new Promise(async (resolve, reject) => {
+      LoginWithAmazon.logout((error) => {
+        if(error) {
+          dispatch({ type: 'ERROR', error: error })
```
Comment (BryceV) [inline src/actions/index.js:9]: Is there a better way to handle errors? This is used throughout the code.

File: src/screens/Auth/Login.js [hunk 6]
```diff
@@ -39,6 +57,15 @@ export default class Login extends Component {
     return (
       <Wallpaper style={{padding: '10%'}}>
         <Logo />
+        <AmazonSignInButton 
+          onPress={this.loginWithAmazon} 
+          spinning={this.state.amazonWaiting}
+        />
+        <View style={styles.socialDiv}>
+          <Divider />
+          <Text style={styles.divText}>or</Text>
```
Comment (BryceV) [inline src/screens/Auth/Login.js:66]: Lets specify this is an alt for artisans only. We can put that message here or around this divider

**Pull request merged**